pkg/skaffold/schema/latest/config.go

```diff
@@ -551,6 +551,27 @@ type ArtifactType struct {
 
 	// KanikoArtifact *alpha* builds images using [kaniko](https://github.com/GoogleContainerTools/kaniko).
 	KanikoArtifact *KanikoArtifact `yaml:"kaniko,omitempty" yamltags:"oneOf=artifact"`
+
+	// CustomArtifact *alpha* builds images using a custom build script written by the user.
+	CustomArtifact *CustomArtifact `yaml:"custom,omitempty" yamltags:"oneOf=artifact"`
+}
+
+// CustomArtifact *alpha* describes an artifact built from a custom build script
+// written by the user. It can be used to build images with builders that aren't directly integrated with skaffold.
+type CustomArtifact struct {
+	// BuildCommand is the command executed to build the image.
+	BuildCommand string `yaml:"buildCommand,omitempty"`
+	// Dependencies are the file dependencies that skaffold should watch for this artifact.
+	Dependencies *CustomDependencies `yaml:"dependencies,omitempty"`
+}
+
+// CustomDependencies is used to specify dependencies for an artifact built by a custom build script.
+type CustomDependencies struct {
+	// Paths should be set to the dependencies for this artifact if no Dockerfile exists.
```

Comment:
I don't understand the "if no Dockerfile exists" part, do we need it here?

Reply:
i think this is for future? the DD talks about `docker`, `jib` dependencies 
```
dependencies: # Only one of the following would be allowed
         dockerfile: path/to/Dockerfile
         bazel: //:myTarget
         jib: /idk/something
         command: query query query woohoo dependencies
```
Do you want to add those once they are supported @priyawadhwa

Reply:
Yup updated the comment, that wasn't meant to be there yet!